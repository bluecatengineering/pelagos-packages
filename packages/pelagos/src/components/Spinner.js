import PropTypes from 'prop-types';
import {t} from '@bluecateng/l10n.macro';

import './Spinner.less';

/** A spinner. */
const Spinner = ({className, size = 'medium', role = 'status', 'aria-label': ariaLabel = t`Loading`, ...props}) => (
	<div
		{...props}
		className={'Spinner CssSpinner Spinner--' + size + (className ? ' ' + className : '')}
		role={role}
		aria-label={ariaLabel}
	/>
);

Spinner.propTypes = {
	/** The component class name(s). */
	className: PropTypes.string,
	/** The spinner size. */
	size: PropTypes.oneOf(['tiny', 'small', 'medium', 'large']),
	/** The element role. */
	role: PropTypes.string,
	/** The element's aria label. */
	'aria-label': PropTypes.string,
};

export default Spinner;
