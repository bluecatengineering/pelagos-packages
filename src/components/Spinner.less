.Spinner {
	position: absolute;
	top: 50%;
	align-self: center;
}

.Spinner--tiny {
	font-size: 4px;
}

.Spinner--small {
	font-size: 7px;
}

.Spinner--medium {
	font-size: 10px;
}

.Spinner--large {
	font-size: 14px;
}
