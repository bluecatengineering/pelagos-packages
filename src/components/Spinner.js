import React from 'react';
import PropTypes from 'prop-types';

import './Spinner.less';

const Spinner = ({className, size, ...props}) => (
	<div {...props} className={'Spinner CssSpinner Spinner--' + size + (className ? ' ' + className : '')} />
);

Spinner.propTypes = {
	className: PropTypes.string,
	size: PropTypes.oneOf(['tiny', 'small', 'medium', 'large']),
};

Spinner.defaultProps = {
	size: 'medium',
};

export default Spinner;
