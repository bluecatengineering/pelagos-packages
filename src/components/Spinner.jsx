import React from 'react';
import PropTypes from 'prop-types';

import './Spinner.less';

const Spinner = ({size, ...props}) => <div {...props} className={'Spinner CssSpinner Spinner--' + size} />;

Spinner.propTypes = {
	size: PropTypes.oneOf(['tiny', 'small', 'medium', 'large']),
};

Spinner.defaultProps = {
	size: 'medium',
};

export default Spinner;
