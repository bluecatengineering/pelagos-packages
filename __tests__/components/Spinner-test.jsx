import React from 'react';
import {shallow} from 'enzyme';

import Spinner from '../../src/components/Spinner';

jest.unmock('../../src/components/Spinner');

describe('Spinner', () => {
	describe('rendering', () => {
		it('renders expected elements', () => {
			const wrapper = shallow(<Spinner id="test" />);
			expect(wrapper.getElement()).toMatchSnapshot();
		});
	});
});
