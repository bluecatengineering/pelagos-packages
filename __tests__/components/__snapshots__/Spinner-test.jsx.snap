// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner rendering renders expected elements 1`] = `
<div
  className="Spinner CssSpinner Spinner--medium"
  id="test"
/>
`;
